opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -14,124 +14,198 @@
 
 from abc import ABC, abstractmethod
 from collections import OrderedDict
+from enum import IntEnum
 from logging import getLogger
 from math import inf
+from threading import Lock
+from typing import Generic, Optional, Sequence, TypeVar
 
-from opentelemetry._metrics.instrument import _Monotonic
+from opentelemetry.sdk._metrics.measurement import Measurement
+from opentelemetry.sdk._metrics.point import Gauge, Histogram, PointT, Sum
 from opentelemetry.util._time import _time_ns
 
+
+class AggregationTemporality(IntEnum):
+    UNSPECIFIED = 0
+    DELTA = 1
+    CUMULATIVE = 2
+
+
+_PointVarT = TypeVar("_PointVarT", bound=PointT)
+
 _logger = getLogger(__name__)
 
 
-class Aggregation(ABC):
-    @property
-    def value(self):
-        return self._value  # pylint: disable=no-member
+class _InstrumentMonotonicityAwareAggregation:
```

Comment:
Might be a stupid question but what is the purpose of keeping track of if instrument/aggregation is monotonic? Is it error checking?

Reply:
Some aggregations need to do special things if the corresponding instrument is monotonic, HistogramAggregation for example needs to keep a sum if the corresponding instrument is monotonic.